src/Workspaces/Core/Desktop/Workspace/SQLite/SQLitePersistentStorage.cs

```diff
@@ -212,6 +216,9 @@ public override void Close()
                     connection.Close_OnlyForUseBySqlPersistentStorage();
                 }
             }
+
+            // let the lock go
+            _dbOwnershipLock.Dispose();
```

Comment:
Can we put this in a try/finally around the entire method.  Or write this like:

```c#
public override void Close()
{
    try
    {
        CloseWorker();
    }
    finally
    {
        ownershipLock.Dispose();
    }
}

private void CloseWorker();
{
             // Flush all pending writes so that all data our features wanted written
             // are definitely persisted to the DB.
             try
             {
                 FlushAllPendingWritesAsync(CancellationToken.None).Wait();
             }
             catch (Exception e)
             {
                 // Flushing may fail.  We still have to close all our connections.
                 StorageDatabaseLogger.LogException(e);
             }
 
             lock (_connectionGate)
             {
                 // Notify any outstanding async work that it should stop.
                 _shutdownTokenSource.Cancel();
 
                 // Go through all our pooled connections and close them.
                 while (_connectionsPool.Count > 0)
                 {
                     var connection = _connectionsPool.Pop();
                      connection.Close_OnlyForUseBySqlPersistentStorage();
                  }
              }
    }
```